lib/ansible/plugins/lookup/azure_keyvault_secret.py

```diff
@@ -0,0 +1,123 @@
+# python 3 headers, required if submitting to Ansible
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: azure_keyvault_secret
+    author:
+        - Hai Cao <t-haicao@microsoft.com>
+    version_added: 2.6
+    requirements:
+        - requests
+        - azure
+        - msrest
+    short_description: read secret from azure key vault.
+    description:
+      - This lookup returns the content of a secret kept in azure key vault.
+    options:
+        _terms:
+            description: secret name of the secret to retrieve, version can be included like this: secret_name/secret_version, note that if version is not provided, key vault will give the latest version.
+            required: True
+        vault_url:
+            description: url of azure key vault to be retrieved from
+            default: 'azure-key-vault'
+            required: True
+        client_id:
+            description: client_id of service principal that has access to the provided azure key vault
+        key:
+            description: key of service principal provided above
+        tenant_id:
+            description: tenant_id of service principal provided above
+    notes:
+        - If this plugin is called on an azure virtual machine and the machine has access to the desired key vault via MSI, then you don't need to provide client_id, key, tenant_id.
```

Comment:
If ansible is running on Azure Virtual Machine with MSI enabled, client_id, secret and tenant isn't necessary. For how to enable MSI on Azure Virutal Machine, please refer to this doc https://docs.microsoft.com/en-us/azure/active-directory/managed-service-identity/